compiler-rt/lib/dfsan/dfsan.cpp

```diff
@@ -143,21 +163,39 @@ SANITIZER_INTERFACE_ATTRIBUTE uptr __dfsan_shadow_ptr_mask;
 // | reserved by kernel |
 // +--------------------+ 0x0000000000
 
-typedef atomic_dfsan_label dfsan_union_table_t[kNumLabels][kNumLabels];
-
+//typedef atomic_dfsan_label dfsan_union_table_t[kNumLabels][kNumLabels];
+typedef atomic_dfsan_label old_union_table_t[oldNumLabels][oldNumLabels];
+//typedef atomic_dfsan_label dfsan_row[kNumLabels];
 #ifdef DFSAN_RUNTIME_VMA
 // Runtime detected VMA size.
 int __dfsan::vmaSize;
 #endif
 
 static uptr UnusedAddr() {
-  return UnionTableAddr() + sizeof(dfsan_union_table_t);
+  return UnionTableAddr() + sizeof(old_union_table_t);
```

Comment:
@ESultanik You can see here where the union table size is used to partition a region of the mmaped space. We technically don't need this space anymore, but it should not really affect us. If you look in dfsan_platform.h, I already bumped up the address space to support all the shadow memory we want.